server/src/main/java/pl/edu/agh/RaftServer.java

```diff
@@ -86,16 +94,29 @@ public State getState() {
     private TcpServer<ByteBuf, ByteBuf> createTcpServer(int port) {
         TcpServer<ByteBuf, ByteBuf> tcpServer = TcpServer.newServer(port);
         tcpServer.enableWireLogging("server", LogLevel.DEBUG)
-                .start(connection -> connection.writeBytesAndFlushOnEach(connection.getInput()
-                        .map(MessageUtils::toObject)
-                        .map(this::handleRequest)
-                        .filter(Optional::isPresent)
-                        .map(Optional::get)
-                        .map(SerializationUtils::serialize)
-                ));
+                .start(connection -> {
+                            checkIfClient(connection);
+                            return connection.writeStringAndFlushOnEach(connection.getInput()
+                                    .map(bb -> bb.toString(Charset.defaultCharset()))
```

Comment:
Oh we are handling strings. My biggest bug was resolved by switching from serializing objects to byte arrays to serializing them to strings.